sphinx/builders/__init__.py

```diff
@@ -420,8 +421,22 @@ def read(self) -> list[str]:
             self._read_serial(docnames)
 
         if self.config.root_doc not in self.env.all_docs:
-            raise SphinxError('root file %s not found' %
-                              self.env.doc2path(self.config.root_doc))
+            root_doc_path = self.env.doc2path(self.config.root_doc)
+            if ("**" in self.config.exclude_patterns or
```

Comment:
I think using `fnmatch` is sufficient. No need to have the `**` check IMO. I'll add tests (in the `tests/` folder) this w-e directly on your branch and will merge it afterwards.